src/ripple/conditions/hashlock.h

```diff
@@ -0,0 +1,112 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_CONDITIONS_HASHLOCK_H
+#define RIPPLE_CONDITIONS_HASHLOCK_H
+
+#include <ripple/basics/base_uint.h>
+#include <ripple/conditions/condition.h>
+#include <ripple/conditions/fulfillment.h>
+#include <ripple/conditions/impl/base64.h>
+#include <ripple/protocol/digest.h>
+#include <boost/regex.hpp>
+#include <boost/optional.hpp>
+#include <algorithm>
+#include <iterator>
+#include <vector>
+
+namespace ripple {
+namespace cryptoconditions {
+
+struct hashlock_t final
+    : public fulfillment_t
+{
+    Buffer payload_;
+
+public:
+    hashlock_t() = default;
+
+    hashlock_t (Slice s)
+        : payload_ (s)
+    {
+        // TODO: We don't want to throw. Devise better
+        //       interface for constructing hashlock from
+        //       given buffer.
```

Comment:
This constructor is only used in testing. As such, let's remove the exception and add a `bool&` parameter that says if there's an error or not and document this constructor is for testing only (there will be a commit to cherry pick if you want this change in the summary comment).